pynapple/io/interface_nwb.py

```diff
@@ -458,9 +458,27 @@ def __getitem__(self, key):
             If key is not in the dictionary
         """
         if key.__hash__:
+            if "/" in key:
+                # allow user to specify the full path to the object
+                # store the original key so we can verify it later
+                original_key = key
+                if original_key.startswith("/"):
+                    # remove leading slash
+                    original_key = original_key[1:]
+                # use just the last part of the key
+                key = key.split("/")[-1]
+
+            else:
+                original_key = None
             if self.__contains__(key):
                 if isinstance(self.data[key], dict) and "id" in self.data[key]:
-                    obj = self.nwb.objects[self.data[key]["id"]]
+                    obj_id = self.data[key]["id"]
+                    obj = self.nwb.objects[obj_id]
+                    if original_key is not None:
+                        if _path_for_nwb_object(obj) != original_key:
+                            raise KeyError(
```

Comment:
Is it possible to write a little tests for this? I am not sure I understand the case where this would happen.